pkg/utils/poller/poller.go

```diff
@@ -0,0 +1,91 @@
+package poller
+
+import (
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/mailbox"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+// Poller is a component that polls a function at a given interval
+// and delivers the result to subscribers
+type Poller[T any] struct {
+	services.StateMachine
+	pollingInterval time.Duration
+	pollingFunc     func() (T, error)
+	logger          logger.Logger
+
+	subscribers []*mailbox.Mailbox[T]
```

Comment:
We should not use mailbox as it drops old messages, if we exceed queue capacity. Each instance of the Poller should be used by single subscriber. If subscriber does not read msg from the channel, it's fine to block the polling goroutine